staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go

```diff
@@ -74,3 +77,21 @@ func (lm *lockMap) unlockEntry(entry string) {
 func getContextWithCancel() (context.Context, context.CancelFunc) {
 	return context.WithCancel(context.Background())
 }
+
+// aggregateGoroutinesWithDelay aggregates goroutines and runs them
+// in parallel with delay before starting each goroutine
+func aggregateGoroutinesWithDelay(delay time.Duration, funcs ...func() error) utilerrors.Aggregate {
+	errChan := make(chan error, len(funcs))
+
+	for _, f := range funcs {
+		go func(f func() error) { errChan <- f() }(f)
+		time.Sleep(delay)
```

Comment:
Could we add some jiffies to the delay? in case the default vmssVMInstanceUpdateDelay is not enough for some cases

Reply:
@feiskyer We delay launching the goroutines but don't actual see if there is an error until we finish the aggregation of routines. Do you have something in mind?

Reply:
OK, let's keep it as is